### PR TITLE
azure web deploy update

### PR DESCRIPTION
**Pull request by yaananth**

### CODE REVIEW

File: Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1 [hunk 1]
```diff
@@ -121,7 +121,7 @@ Invoke-Expression -Command $finalCommand
 #Update Deployment status - https://github.com/projectkudu/kudu/wiki/REST-API#deployment
 
 if($azureWebSite) {
-    $matchedWebSiteName = $azureWebSite.EnabledHostNames | Where-Object { $_ -like '*.scm.azurewebsites.net*' } | Select-Object -First 1
+    $matchedWebSiteName = $azureWebSite.EnabledHostNames | Where-Object { $_ -like '*.scm*azurewebsites.net*' } | Select-Object -First 1
```
Comment (yaananth) [inline Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1]: for app service environments, scm site would be scm.blahblah.p.azurewebsites.net

File: Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1 [hunk 2]
```diff
@@ -133,7 +133,17 @@ if($azureWebSite) {
...
-        $deploymentId = Get-TaskVariable $distributedTaskContext "build.sourceVersion" #let's use commitId as unique deploymentId
+        $deploymentId = Get-TaskVariable $distributedTaskContext "build.sourceVersion" #let's use commitId as unique deploymentId in build context
+        if([string]::IsNullOrEmpty($deploymentId)) {
+            $deploymentId = Get-TaskVariable $distributedTaskContext "release.releaseUri" #let's use releaseUri as unique deploymentId in release context
+        }
+        
+        if([string]::IsNullOrEmpty($deploymentId)) {
+            #No point in proceeding further
+            Write-Warning (Get-LocalizedString -Key "Cannot update deployment status, unique deploymentId cannot be retrieved")  
+            Return
```
Comment (yaananth) [inline Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1]: is Return the write thing to do here?

File: Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1 (hunk 1 above)
Comment (ericsciple) [inline Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1]: Will this produce false positives - or is that not a practical concern? You might want to add a comment if not.

**Comment by ericsciple** [comment]
Looks good.

**Pull request merged**